File: app/src/main/java/xyz/ashu/testssl/MainActivity.java
package xyz.ashu.testssl;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import xyz.ashu.demo11.R;
import xyz.ashu.testssl.network.Api;

public class MainActivity extends AppCompatActivity {

    private TextView mTvHello;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTvHello = findViewById(R.id.tv_hello);
        mTvHello.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getSomething();
            }
        });
    }

    private void getSomething() {
        new Api().getCryptoPercent(new Api.OnResponseListener() {
            @Override
            public void onResponse(String response) {
                refreshView(response);
            }
        });
    }

    private void refreshView(String response) {
        mTvHello.post(new Runnable() {
            @Override
            public void run() {
                mTvHello.setText(response);
            }
        });
    }
}

File: test_tinker/src/main/java/xyz/ashu/test_tinker/MainActivity.kt
package xyz.ashu.test_tinker

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.tencent.bugly.crashreport.CrashReport
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tv_hello.setOnClickListener { tv_hello.text = "原始信息" }
        tv_crash.setOnClickListener { CrashReport.testJavaCrash() }
    }
}
